frontend/ui/uimanager.lua

```diff
@@ -527,7 +527,7 @@ function UIManager:schedule(sched_time, action, ...)
     if e ~= 0 then
         -- Do a binary insert.
         repeat
-            p = math.floor((e + s) / 2) -- Not necessary to use (s + (e -s) / 2) here!
+            p = bit.rshift(e + s, 1) -- Not necessary to use (s + (e -s) / 2) here!
```

Comment:
Should work for small number, but not for big ones (>1e10). 
I think the jit does the same for us, but the code would stay more readable.

Reply:
I've just re-read https://bitop.luajit.org/semantics.html ;).

We're only dealing with array indices here, so we should be well and truly safe.

(I'd be more worried if we were chopping the time field).

Reply:
It also seems like arrays can only hold UINT32_MAX slots anyway, and that works just fine over here ;).

```lua
> print(bit.rshift(4294967295, 1))
2147483647
```